cli/args/flags.rs

```diff
@@ -1849,6 +1850,16 @@ Directory arguments are expanded to all contained files matching the glob
     )
     .arg(no_clear_screen_arg())
     .arg(script_arg().last(true))
+    .arg(
+      Arg::new("junit")
+        .long("junit")
+        .value_name("PATH")
+        .value_hint(ValueHint::FilePath)
+        .help("Write a JUnit XML test report to PATH (default: junit.xml)")
```

Comment:
Maybe instead it should write to stdout by default for `--junit` and do compound reporting only for `--junit=junit.xml`. That aligns better with what we'd want to do for `--json`.

Reply:
I considered that but since the JUnit XML is intended to be serialized at the end of test execution, it would mean that there would be no output at all from the test suite until the very end which doesn't seem like the expected UX 🤔

Reply:
I think that's fine, I actually believe the same should be true of `--json` and streaming JSON lines output should use a different flag like `--jsonl`.

Reply:
I was wondering about this as well. I like @nayeemrmn's suggestion about having it output to stdout if no file path is provided. Maybe in this case it could do pretty reporting on stderr?

Reply:
+1 I'm thinking about tap and dot reporters and in such case that would make more sense for proposed behavior the flags would be exclusive)

Reply:
OK, I'll work on updating this. Looking at it, I think it would be easiest for the PrettyTestReporter to always write on stderr which also seems to better align with where Rust thinks things should be written

> Use println! only for the primary output of your program. Use [eprintln!](https://doc.rust-lang.org/std/macro.eprintln.html) instead to print error and progress messages. 

([source](https://doc.rust-lang.org/std/macro.println.html))

Reply:
OK changing the output stream for `PrettyTestReporter` is rather involved because it requires changes to the integration testing framework. For this PR, I'm going to make it such that only one reporter can output to stdout for the moment and make a TODO

Reply:
> Maybe in this case it could do pretty reporting on stderr?

Like one format on stdout and one on stderr? -1 on that... the first usage is usually reading both streams on the terminal to check out the flag, it would be a strange sight and I'd be surprised if there was precedent for that

Reply:
Yeah I think we can punt this decision until the other reporters are better defined by just disabling pretty if JUnit is writing to stdout.

Reply:
Implemented in [17a6a72](https://github.com/denoland/deno/pull/19747/commits/17a6a72c75c467336b399a6c7c06a3b82efd7638)